Test that makes no sense?

When running the tests against current master under Python 2.7 using the standard `bin/test` script they all complete successfully. However, running them via `tox` produces non-deterministic errors. Example:

```python
Failure in test test_runAllExportSteps_empty (Products.GenericSetup.tests.test_tool.SetupToolTests)
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/Users/jens/src/Products.GenericSetup/Products/GenericSetup/tests/test_tool.py", line 890, in test_runAllExportSteps_empty
    self.assertEqual(result['steps'][1], 'step_registries')
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 506, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: u'toolset' != 'step_registries'
```

The test code (see https://github.com/zopefoundation/Products.GenericSetup/blob/master/Products/GenericSetup/tests/test_tool.py#L882) appears to assume that `result` from a call to `tool.runAllExportSteps()` has a fixed order of steps. But that makes no sense when I look at how the result is put together. In `listExportSteps` the steps are combined from two `set` instances, which obviously don't have a sorting order.

Is the assumption of a fixed order wrong or is the implementation wrong which disregards any order?

--- a/Products/GenericSetup/tests/test_registry.py
+++ b/Products/GenericSetup/tests/test_registry.py
@@ -342,8 +342,7 @@ class ImportStepRegistryTests(
                               dependencies=('four', ))
 
         incomplete = registry.checkComplete()
-        self.assertEqual(len(incomplete), 1)
-        self.failUnless(('two', 'four') in incomplete)
+        self.assertTrue(('two', 'four') in incomplete)
 
     def test_generateXML_empty(self):
 

--- a/Products/GenericSetup/tests/test_registry.py
+++ b/Products/GenericSetup/tests/test_registry.py
@@ -60,8 +60,10 @@ DOC_FUNC_NAME = '%s.%s' % (__name__, DOC_FUNC.__name__)
 #==============================================================================
 
 
-class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, ConformsToIImportStepRegistry
-                              ):
+class ImportStepRegistryTests(
+    BaseRegistryTests,
+    ConformsToIStepRegistry,
+    ConformsToIImportStepRegistry):
 
     layer = ExportImportZCMLLayer
 
@@ -103,8 +105,8 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='docstring', version='1', handler=DOC_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='docstring', version='1', handler=DOC_FUNC,
+                              dependencies=())
 
         info = registry.getStepMetadata('docstring')
         self.assertEqual(info['id'], 'docstring')
@@ -117,8 +119,8 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='docstring', version='1', handler=DOC_FUNC, dependencies=(), title='Title'
-                              )
+        registry.registerStep(id='docstring', version='1', handler=DOC_FUNC,
+                              dependencies=(), title='Title')
 
         info = registry.getStepMetadata('docstring')
         self.assertEqual(info['id'], 'docstring')
@@ -131,8 +133,9 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', 'three'), title='One Step', description='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', 'three'),
+                              title='One Step', description='One small step')
 
         steps = registry.listSteps()
         self.assertEqual(len(steps), 1)
@@ -157,8 +160,8 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry.registerStep(id='one', version='1', handler=ONE_FUNC_NAME)
 
-        self.assertRaises(KeyError, registry.registerStep, id='one', version='0', handler=ONE_FUNC
-                          )
+        self.assertRaises(KeyError, registry.registerStep, id='one',
+                          version='0', handler=ONE_FUNC)
 
         registry.registerStep(id='one', version='1', handler=ONE_FUNC_NAME)
 
@@ -169,11 +172,13 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', 'three'), title='One Step', description='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', 'three'),
+                              title='One Step', description='One small step')
 
-        registry.registerStep(id='one', version='1.1', handler=ONE_FUNC, dependencies=(), title='Leads to Another', description='Another small step'
-                              )
+        registry.registerStep(id='one', version='1.1', handler=ONE_FUNC,
+                              dependencies=(), title='Leads to Another',
+                              description='Another small step')
 
         info = registry.getStepMetadata('one')
         self.assertEqual(info['id'], 'one')
@@ -186,14 +191,14 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=())
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=())
 
-        registry.registerStep(id='three', version='3', handler=THREE_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='three', version='3', handler=THREE_FUNC,
+                              dependencies=())
 
         steps = registry.listSteps()
         self.assertEqual(len(steps), 3)
@@ -205,11 +210,11 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', )
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', ))
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=())
 
         steps = registry.sortSteps()
         self.assertEqual(len(steps), 2)
@@ -222,14 +227,14 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('three', ), title='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('three', ), title='One small step')
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=('one', ), title='Texas two step'
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=('one', ), title='Texas two step')
 
-        registry.registerStep(id='three', version='3', handler=THREE_FUNC, dependencies=(), title='Gimme three steps'
-                              )
+        registry.registerStep(id='three', version='3', handler=THREE_FUNC,
+                              dependencies=(), title='Gimme three steps')
 
         steps = registry.sortSteps()
         self.assertEqual(len(steps), 3)
@@ -243,17 +248,18 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', ), title='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', ), title='One small step')
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=('four', ), title='Texas two step'
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=('four', ), title='Texas two step')
 
-        registry.registerStep(id='three', version='3', handler=THREE_FUNC, dependencies=('four', ), title='Gimme three steps'
-                              )
+        registry.registerStep(id='three', version='3', handler=THREE_FUNC,
+                              dependencies=('four', ),
+                              title='Gimme three steps')
 
-        registry.registerStep(id='four', version='4', handler=FOUR_FUNC, dependencies=(), title='Four step program'
-                              )
+        registry.registerStep(id='four', version='4', handler=FOUR_FUNC,
+                              dependencies=(), title='Four step program')
 
         steps = registry.sortSteps()
         self.assertEqual(len(steps), 4)
@@ -269,17 +275,19 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', 'three'), title='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', 'three'),
+                              title='One small step')
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=('four', ), title='Texas two step'
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=('four', ), title='Texas two step')
 
-        registry.registerStep(id='three', version='3', handler=THREE_FUNC, dependencies=('four', ), title='Gimme three steps'
-                              )
+        registry.registerStep(id='three', version='3', handler=THREE_FUNC,
+                              dependencies=('four', ),
+                              title='Gimme three steps')
 
-        registry.registerStep(id='four', version='4', handler=FOUR_FUNC, dependencies=(), title='Four step program'
-                              )
+        registry.registerStep(id='four', version='4', handler=FOUR_FUNC,
+                              dependencies=(), title='Four step program')
 
         steps = registry.sortSteps()
         self.assertEqual(len(steps), 4)
@@ -295,15 +303,15 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', )
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', ))
 
         incomplete = registry.checkComplete()
         self.assertEqual(len(incomplete), 1)
         self.failUnless(('one', 'two') in incomplete)
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=())
 
         self.assertEqual(len(registry.checkComplete()), 0)
 
@@ -311,28 +319,27 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne()
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', 'three')
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', 'three'))
 
         incomplete = registry.checkComplete()
         self.assertEqual(len(incomplete), 2)
         self.failUnless(('one', 'two') in incomplete)
         self.failUnless(('one', 'three') in incomplete)
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=())
 
         incomplete = registry.checkComplete()
-        self.assertEqual(len(incomplete), 1)
-        self.failUnless(('one', 'three') in incomplete)
+        self.assertEqual(list(incomplete), [('one', 'three')])
 
-        registry.registerStep(id='three', version='3', handler=THREE_FUNC, dependencies=()
-                              )
+        registry.registerStep(id='three', version='3', handler=THREE_FUNC,
+                              dependencies=())
 
-        self.assertEqual(len(registry.checkComplete()), 0)
+        self.assertEqual(list(registry.checkComplete()), [])
 
-        registry.registerStep(id='two', version='2.1', handler=TWO_FUNC, dependencies=('four', )
-                              )
+        registry.registerStep(id='two', version='2.1', handler=TWO_FUNC,
+                              dependencies=('four', ))
 
         incomplete = registry.checkComplete()
         self.assertEqual(len(incomplete), 1)
@@ -348,8 +355,9 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=(), title='One Step', description='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=(), title='One Step',
+                              description='One small step')
 
         self._compareDOM(registry.generateXML(), _SINGLE_IMPORT_XML)
 
@@ -357,14 +365,17 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=('two', ), title='One Step', description='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=('two', ), title='One Step',
+                              description='One small step')
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=('three', ), title='Two Steps', description='Texas two step'
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=('three', ), title='Two Steps',
+                              description='Texas two step')
 
-        registry.registerStep(id='three', version='3', handler=THREE_FUNC, dependencies=(), title='Three Steps', description='Gimme three steps'
-                              )
+        registry.registerStep(id='three', version='3', handler=THREE_FUNC,
+                              dependencies=(), title='Three Steps',
+                              description='Gimme three steps')
 
         self._compareDOM(registry.generateXML(), _ORDERED_IMPORT_XML)
 
@@ -372,8 +383,8 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='one', version='1', handler=ONE_FUNC, dependencies=(), description='One small step'
-                              )
+        registry.registerStep(id='one', version='1', handler=ONE_FUNC,
+                              dependencies=(), description='One small step')
 
         info_list = registry.parseXML(_EMPTY_IMPORT_XML)
 
@@ -383,8 +394,9 @@ class ImportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='two', version='2', handler=TWO_FUNC, dependencies=(), title='Two Steps', description='Texas two step'
-                              )
+        registry.registerStep(id='two', version='2', handler=TWO_FUNC,
+                              dependencies=(), title='Two Steps',
+                              description='Texas two step')
 
         info_list = registry.parseXML(_SINGLE_IMPORT_XML)
 
@@ -448,8 +460,10 @@ _ORDERED_IMPORT_XML = """\
 #==============================================================================
 
 
-class ExportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, ConformsToIExportStepRegistry
-                              ):
+class ExportStepRegistryTests(
+    BaseRegistryTests,
+    ConformsToIStepRegistry,
+    ConformsToIExportStepRegistry):
 
     layer = ExportImportZCMLLayer
 
@@ -544,8 +558,8 @@ class ExportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='one', handler=ONE_FUNC, title='One Step', description='One small step'
-                              )
+        registry.registerStep(id='one', handler=ONE_FUNC, title='One Step',
+                              description='One small step')
 
         self._compareDOM(registry.generateXML(), _SINGLE_EXPORT_XML)
 
@@ -553,14 +567,15 @@ class ExportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='one', handler=ONE_FUNC, title='One Step', description='One small step'
-                              )
+        registry.registerStep(id='one', handler=ONE_FUNC, title='One Step',
+                              description='One small step')
 
-        registry.registerStep(id='two', handler=TWO_FUNC, title='Two Steps', description='Texas two step'
-                              )
+        registry.registerStep(id='two', handler=TWO_FUNC, title='Two Steps',
+                              description='Texas two step')
 
-        registry.registerStep(id='three', handler=THREE_FUNC, title='Three Steps', description='Gimme three steps'
-                              )
+        registry.registerStep(id='three', handler=THREE_FUNC,
+                              title='Three Steps',
+                              description='Gimme three steps')
 
         self._compareDOM(registry.generateXML(), _ORDERED_EXPORT_XML)
 
@@ -568,7 +583,8 @@ class ExportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='one', handler=ONE_FUNC, description='One small step'
+        registry.registerStep(id='one', handler=ONE_FUNC,
+                              description='One small step'
                               )
 
         info_list = registry.parseXML(_EMPTY_EXPORT_XML)
@@ -579,7 +595,8 @@ class ExportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='two', handler=TWO_FUNC, title='Two Steps', description='Texas two step'
+        registry.registerStep(id='two', handler=TWO_FUNC, title='Two Steps',
+                              description='Texas two step'
                               )
 
         info_list = registry.parseXML(_SINGLE_EXPORT_XML)
@@ -596,7 +613,8 @@ class ExportStepRegistryTests(BaseRegistryTests, ConformsToIStepRegistry, Confor
 
         registry = self._makeOne().__of__(self.app)
 
-        registry.registerStep(id='two', handler=TWO_FUNC, title='Two Steps', description='Texas two step'
+        registry.registerStep(id='two', handler=TWO_FUNC, title='Two Steps',
+                              description='Texas two step'
                               )
 
         info_list = registry.parseXML(_SINGLE_EXPORT_XML, encoding='ascii')
@@ -723,8 +741,11 @@ class ToolsetRegistryTests(BaseRegistryTests, ConformsToIToolsetRegistry
 
     def test_addRequiredTool_multiple(self):
 
-        REQUIRED = (('one', 'path.to.one'), ('two', 'path.to.two'), ('three', 'path.to.three')
-                    )
+        REQUIRED = (
+            ('one', 'path.to.one'),
+            ('two', 'path.to.two'),
+            ('three', 'path.to.three'),
+        )
 
         site = self._initSite()
         configurator = self._makeOne().__of__(site)
@@ -907,8 +928,8 @@ class ProfileRegistryTests(BaseRegistryTests, ConformsToIProfileRegistry
         PROFILE_ID = 'TestProduct:one'
 
         registry = self._makeOne()
-        registry.registerProfile(NAME, TITLE, DESCRIPTION, PATH, PRODUCT, PROFILE_TYPE
-                                 )
+        registry.registerProfile(NAME, TITLE, DESCRIPTION, PATH, PRODUCT,
+                                 PROFILE_TYPE)
 
         self.assertEqual(len(registry.listProfiles()), 1)
         self.assertEqual(len(registry.listProfileInfo()), 1)
@@ -975,8 +996,8 @@ class ProfileRegistryTests(BaseRegistryTests, ConformsToIProfileRegistry
         DERIVED_FOR = IDerivedSite
 
         registry = self._makeOne()
-        registry.registerProfile(NAME, TITLE, DESCRIPTION, PATH, PRODUCT, PROFILE_TYPE, for_=FOR
-                                 )
+        registry.registerProfile(NAME, TITLE, DESCRIPTION, PATH, PRODUCT,
+                                 PROFILE_TYPE, for_=FOR)
 
         self.assertEqual(len(registry.listProfiles()), 1)
         self.assertEqual(len(registry.listProfiles(for_=FOR)), 1)
@@ -997,8 +1018,8 @@ class ProfileRegistryTests(BaseRegistryTests, ConformsToIProfileRegistry
         self.assertEqual(info1, info3)
 
         # ...and that this one fails.
-        self.assertRaises(KeyError, registry.getProfileInfo, PROFILE_ID, for_=NOT_FOR
-                          )
+        self.assertRaises(KeyError, registry.getProfileInfo, PROFILE_ID,
+                          for_=NOT_FOR)
 
         info = registry.getProfileInfo(PROFILE_ID, for_=FOR)
 

--- a/Products/GenericSetup/tests/test_tool.py
+++ b/Products/GenericSetup/tests/test_tool.py
@@ -886,10 +886,10 @@ class SetupToolTests(FilesystemTestBase, TarballTester, ConformsToISetupTool):
 
         result = tool.runAllExportSteps()
 
-        self.assertEqual(len(result['steps']), 4)
-        self.assertEqual(result['steps'][1], 'step_registries')
-        self.assertEqual(result['messages']['step_registries'], None
-                         )
+        self.assertEqual(
+            sorted(result['steps']),
+            ['componentregistry', 'rolemap', 'step_registries', 'toolset'])
+        self.assertEqual(result['messages']['step_registries'], None)
 
     def test_runAllExportSteps_default(self):
         site = self._makeSite()
@@ -902,8 +902,9 @@ class SetupToolTests(FilesystemTestBase, TarballTester, ConformsToISetupTool):
 
         result = tool.runAllExportSteps()
 
-        self.assertEqual(len(result['steps']), 4)
-        self.assertEqual(result['steps'][1], 'step_registries')
+        self.assertEqual(sorted(result['steps']),
+                         ['componentregistry', 'rolemap',
+                          'step_registries', 'toolset'])
         self.assertEqual(result['messages']['step_registries'], None
                          )
         fileish = StringIO(result['tarball'])
@@ -936,14 +937,13 @@ class SetupToolTests(FilesystemTestBase, TarballTester, ConformsToISetupTool):
         result = tool.runAllExportSteps()
 
         self.assertEqual(len(result['steps']), 5)
+        self.assertEqual(sorted(result['steps']),
+                         ['componentregistry', 'properties', 'rolemap',
+                          'step_registries', 'toolset'])
 
-        self.failUnless('properties' in result['steps'])
-        self.assertEqual(result['messages']['properties'], 'Exported properties'
-                         )
-
-        self.failUnless('step_registries' in result['steps'])
-        self.assertEqual(result['messages']['step_registries'], None
-                         )
+        self.assertEqual(result['messages']['properties'],
+                         'Exported properties')
+        self.assertEqual(result['messages']['step_registries'], None)
 
         fileish = StringIO(result['tarball'])
 
@@ -1018,25 +1018,29 @@ class SetupToolTests(FilesystemTestBase, TarballTester, ConformsToISetupTool):
         self.assertTrue('Manager' in site.valid_roles())
 
     def test_createSnapshot_default(self):
-        _EXPECTED = [('import_steps.xml', _DEFAULT_STEP_REGISTRIES_IMPORT_XML),
-                     ('export_steps.xml', _DEFAULT_STEP_REGISTRIES_EXPORT_XML),
-                     ('rolemap.xml', 'dummy'),
-                     ('toolset.xml', 'dummy')]
+        _EXPECTED = [
+            ('import_steps.xml', _DEFAULT_STEP_REGISTRIES_IMPORT_XML),
+            ('export_steps.xml', _DEFAULT_STEP_REGISTRIES_EXPORT_XML),
+            ('rolemap.xml', 'dummy'),
+            ('toolset.xml', 'dummy'),
+        ]
 
         site = self._makeSite()
         site.setup_tool = self._makeOne('setup_tool')
         tool = site.setup_tool
         tool._import_registry.registerStep('foo', handler='foo.bar')
-        tool._export_registry.registerStep('step_registries',
-                                           'Products.GenericSetup.tool.exportStepRegistries',
-                                           'Export import / export steps.')
+        tool._export_registry.registerStep(
+            'step_registries',
+            'Products.GenericSetup.tool.exportStepRegistries',
+            'Export import / export steps.')
 
         self.assertEqual(len(tool.listSnapshotInfo()), 0)
 
         result = tool.createSnapshot('default')
 
-        self.assertEqual(len(result['steps']), 4)
-        self.assertEqual(result['steps'][1], 'step_registries')
+        self.assertEqual(
+            sorted(result['steps']),
+            ['componentregistry', 'rolemap', 'step_registries', 'toolset'])
         self.assertEqual(result['messages']['step_registries'], None
                          )
 

